src/main/java/org/jenkinsci/plugins/configfiles/GlobalConfigFiles.java

```diff
@@ -21,6 +21,7 @@
 public class GlobalConfigFiles extends Descriptor<GlobalConfigFiles> implements ConfigFileStore, ExtensionPoint, Describable<GlobalConfigFiles> {
 
     private Collection<Config> configs = new ArrayList<Config>();
+    private transient Map<String, Config> configsById = new HashMap<String, Config>();
```

Comment:
I did a few short tests, and I think the same can be done by just replacing config definition with this:

```
    private transient Comparator<Config> COMPARATOR = new Comparator<Config>() {
        @Override
        public int compare(Config o1, Config o2) {
            return o1.id.compareTo(o2.id);
        }
    };

    private Collection<Config> configs = new TreeSet<>(COMPARATOR);
```

...but I'm not sure yet if this is better or not